plugins/module_utils/linode_common.py

```diff
@@ -141,6 +156,17 @@ def register_action(self, description: str) -> None:
         self.results['changed'] = True
         self.results['actions'].append(description)
 
+    def _get_resource_by_id(self, resource_type: Type[LinodeAPIType], resource_id: int):
+        try:
+            resource = resource_type(self.client, resource_id)
+            resource._api_get()
+            return resource
+        except Exception as exception:
+            resource_name = RESOURCE_NAMES.get(resource_type) or type(resource_type).__name__
```

Comment:
Nice! I wonder if it would be worth excluding the hard-coded `RESOURCE_NAMES` altogether so the function can be fully generic

Reply:
I am not sure about that. But the hardcoded dict does make the error message more clear by providing the names. In the future, we can put a `name` attribute in the API object of the Python API wrapper.

Reply:
Gotcha 👍 
I definitely think having a `display_name` attribute on resource classes in `linode_api4-python` would be the best approach for this going forward.